lib/web_ui/lib/src/engine/pointer_converter.dart

```diff
@@ -0,0 +1,255 @@
+      timeStamp: datum.timeStamp,
+      change: datum.change,
+      kind: datum.kind,
+      signalKind: datum.signalKind,
+      device: datum.device,
+      pointerIdentifier: state.pointer ?? 0,
+      physicalX: datum.physicalX,
+      physicalY: datum.physicalY,
+      physicalDeltaX: deltaX,
+      physicalDeltaY: deltaY,
+      buttons: datum.buttons,
+      obscured: datum.obscured,
+      synthesized: datum.synthesized,
+      pressure: datum.pressure,
+      pressureMin: datum.pressureMin,
+      pressureMax: datum.pressureMax,
+      distance: datum.distance,
+      distanceMax: datum.distanceMax,
+      size: datum.size,
+      radiusMajor: datum.radiusMajor,
+      radiusMinor: datum.radiusMinor,
+      radiusMin: datum.radiusMin,
+      radiusMax: datum.radiusMax,
+      orientation: datum.orientation,
+      tilt: datum.tilt,
+      platformData: datum.platformData,
+      scrollDeltaX: datum.scrollDeltaX,
+      scrollDeltaY: datum.scrollDeltaY,
+    );
+  }
+
+  bool _locationHasChanged(ui.PointerData datum) {
+    assert(_pointers.containsKey(datum.device));
+    final _PointerState state = _pointers[datum.device];
+    return state.x != datum.physicalX || state.y != datum.physicalY;
+  }
+
+  ui.PointerData _synthesizeFrom(ui.PointerData datum, ui.PointerChange change) {
+    assert(_pointers.containsKey(datum.device));
+    final _PointerState state = _pointers[datum.device];
+    final double deltaX = datum.physicalX - state.x;
+    final double deltaY = datum.physicalY - state.y;
+    state.x = datum.physicalX;
+    state.y = datum.physicalY;
+    return ui.PointerData(
+      timeStamp: datum.timeStamp,
+      change: change,
+      kind: datum.kind,
+      // All the pointer data except scroll should not have a signal kind, and
+      // there is no use case for synthetic scroll event. We should be
+      // safe to default it to ui.PointerSignalKind.none.
+      signalKind: ui.PointerSignalKind.none,
+      device: datum.device,
+      pointerIdentifier: state.pointer ?? 0,
+      physicalX: datum.physicalX,
+      physicalY: datum.physicalY,
+      physicalDeltaX: deltaX,
+      physicalDeltaY: deltaY,
+      buttons: datum.buttons,
+      obscured: datum.obscured,
+      synthesized: true,
+      pressure: datum.pressure,
+      pressureMin: datum.pressureMin,
+      pressureMax: datum.pressureMax,
+      distance: datum.distance,
+      distanceMax: datum.distanceMax,
+      size: datum.size,
+      radiusMajor: datum.radiusMajor,
+      radiusMinor: datum.radiusMinor,
+      radiusMin: datum.radiusMin,
+      radiusMax: datum.radiusMax,
+      orientation: datum.orientation,
+      tilt: datum.tilt,
+      platformData: datum.platformData,
+      scrollDeltaX: datum.scrollDeltaX,
+      scrollDeltaY: datum.scrollDeltaY,
+    );
+  }
+
+  /// Convert the given list pointer data into a sequence of framework-compatible
+  /// pointer data.
+  Iterable<ui.PointerData> convert(List<ui.PointerData> data) sync* {
```

Comment:
I understand that this is how the Flutter framework used to implement this. But on the web, `sync*` is expensive and we try to avoid it. Are there any downsides if we change this method to mutate the `data` list in-place instead of generating a new iterable? (we might also rename the method to make it clear that it mutates the list in-place).

Reply:
sounds good

Reply:
sync* is expensive everywhere, if we're doing it on dart:ui it should be removed